client.go

```diff
@@ -11,22 +15,111 @@ import (
 	"github.com/tkhq/go-sdk/pkg/store/local"
 )
 
+const DefaultClientVersion = "go-sdk"
+
+type config struct {
+	apiKey          *apikey.Key
+	clientVersion   string
+	registry        strfmt.Registry
+	transportConfig *client.TransportConfig
+}
+
+type Optfn func(c *config) error
+
+// WithClientVersion overrides the client version used for this API client.
+func WithClientVersion(clientVersion string) Optfn {
+	return func(c *config) error {
+		c.clientVersion = clientVersion
+		return nil
+	}
+}
+
+// WithRegistry sets the registry formats used for this API client.
+func WithRegistry(registry strfmt.Registry) Optfn {
+	return func(c *config) error {
+		c.registry = registry
+		return nil
+	}
+}
+
+// WithTransportConfig sets the TransportConfig used for this API client.
+func WithTransportConfig(transportConfig client.TransportConfig) Optfn {
+	return func(c *config) error {
+		c.transportConfig = &transportConfig
+		return nil
+	}
+}
+
+// WithAPIKey sets the API key used for this API client.
+// Users would normally use WithAPIKeyName. This offers a lower-level custom API
+// key.
+func WithAPIKey(apiKey *apikey.Key) Optfn {
+	return func(c *config) error {
+		c.apiKey = apiKey
+		return nil
+	}
+}
+
+// WithAPIKeyName sets the API key to the key loaded from the local keystore
+// with the provided name.
+func WithAPIKeyName(keyname string) Optfn {
+	return func(c *config) error {
+		apiKey, err := local.New[*apikey.Key]().Load(keyname)
+		if err != nil {
+			return errors.Wrap(err, "failed to load API key")
+		}
+		c.apiKey = apiKey
+		return nil
+	}
+}
+
 // New returns a new API Client with the given API key name from the default keystore.
-func New(keyname string) (*Client, error) {
-	apiKey, err := local.New[*apikey.Key, apikey.Metadata]().Load(keyname)
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to load API key")
+func New(options ...Optfn) (*Client, error) {
+	c := &config{
+		clientVersion:   DefaultClientVersion,
+		transportConfig: client.DefaultTransportConfig(),
+	}
+
+	for _, o := range options {
+		o(c)
 	}
+	fmt.Println(c.transportConfig.Host)
+
+	// create transport and client
+	transport := httptransport.New(
+		c.transportConfig.Host,
+		c.transportConfig.BasePath,
+		c.transportConfig.Schemes,
+	)
+	transport.Transport = SetClientVersion(transport.Transport, c.clientVersion)
```

Comment:
This probably deserves a comment to describe what it is doing... maybe "Add client version header".  Either than or rename `SetClientVersion` to something more descriptive... maybe `ClientVersionTransportWrapper()`?